### PR TITLE
feat(windows): custom title bar

### LINKED ISSUE #89: [Windows] Hide/custom title bar like in macOS
**Tell us about the task you want to perform and are unable to do so because the feature is not available**
Increases space for app by roughly 30 pixels vertically

**Describe the solution/feature you'd like us to add**
Add custom title bar, since you are already using `window_manager` anyway, why not also Windows?

**Any other feedback you would like to provide regarding the app**
What I think the app would roughly look like
temp_

### PR DESCRIPTION
**Pull request by damywise**
Follow up from #94 
Fix #89

### CODE REVIEW

File: lib/app.dart [hunk 1]
```diff
@@ -63,22 +69,37 @@ class _DashAppState extends ConsumerState<DashApp> {
         ref.watch(settingsProvider.select((value) => value.isDark));
     return MaterialApp(
       debugShowCheckedModeBanner: false,
-      theme: ThemeData(
-        fontFamily: kFontFamily,
-        fontFamilyFallback: kFontFamilyFallback,
-        colorSchemeSeed: kColorSchemeSeed,
-        useMaterial3: true,
-        brightness: Brightness.light,
+      home: Stack(
```
Comment (damywise) [inline lib/app.dart:72]: What I did here is added the title bar (draggable widget, window controls) above the app so overlays doesn't hinder it.
However, this solution requires another MaterialApp, which hopefully doesn't prove as an issue.

File: lib/main.dart [hunk 2]
```diff
@@ -11,13 +11,14 @@ void main() async {
...
-  } else {
-    var win = getInitialSize();
-    await setupWindow(sz: win.$1, off: win.$2);
   }
   runApp(
     const ProviderScope(
       child: DashApp(),
     ),
   );
+  if (!kIsLinux) {
```
Comment (damywise) [inline lib/main.dart:20]: As before, setting up window after `runApp` to prevent white frame on app startup

**Comment by animator** [comment]
Hi @damywise,

I will test out the PR and add my comments.

File: lib/main.dart (hunk 2 above)
Comment (animator) [reply lib/main.dart:20]: Yes .. I verified it and it resolves the white frame issue.

File: lib/app.dart (hunk 1 above)
Comment (animator) [reply lib/app.dart:72]: For now, I think it will work .. might change it in future when we have a better option.

Comment (animator) [reply lib/app.dart:72]: But, this is leading to an issue in macos layout. So I will look further into it for a fix.

**Comment by animator** [comment]
Comments after review:
Works like a charm on Windows.
leads to an initial layout issue in macos which needs further investigation.

**Comment by damywise** [comment]
Ahh I think I figured out what's wrong with both Windows and macOS.

[This](https://github.com/leanflutter/window_manager/#hidden-at-launch) is not possible in this project specifically because `windows` and `macos` are `.gitignore`d.

Why are they ignored?

**Comment by damywise** [comment]
Sorry that got deleted but basically I said that [this](https://github.com/leanflutter/window_manager/#hidden-at-launch) not being possible to implement is one of the cause.

**Comment by damywise** [comment]
I have tried using it in macos and I don't see any problem, basically.

Also try using profile/release mode, usually in debug mode it's quite worse.

**Comment by animator** [comment]
Thanks @damywise I will take a look at it again.

**Comment by ashitaprasad** [comment]
Hi @damywise 
We are going ahead with the merge. Some fixes were made as window constraints was not working on resize if `setupWindow` was called after `runApp`.
A big thank you for your contribution 🎉  Looking forward to your future contributions to the project. 🙂

**Comment by damywise** [comment]
> window constraints was not working on resize if setupWindow was called after runApp.

Didn't know that was a thing but hey, glad it works now!

(To be honest the only thing that should be after `runApp` is `show()` and `focus()` but for some reason I thought it'd be easier to put the whole `setupWindow` after `runApp`)

**Comment by ashitaprasad** [comment]
> (To be honest the only thing that should be after `runApp` is `show()` and `focus()` but for some reason I thought it'd be easier to put the whole `setupWindow` after `runApp`)

Sure .. I will run a few tests to see if putting `show()` and `focus()` after `runApp` solves the issue.

**Pull request merged**